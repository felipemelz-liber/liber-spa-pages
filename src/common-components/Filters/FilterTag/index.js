export { default } from './FilterTag.jsx';
