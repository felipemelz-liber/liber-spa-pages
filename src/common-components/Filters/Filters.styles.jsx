import styled, { css } from 'styled-components';
import { ModalTag, CardFooter, Button, handleThemeFromObject, TextField } from 'liber-components';

const getTheme = (key, fallback) =>
  css`
    ${({ theme }) => handleThemeFromObject(theme, key, fallback)}
  `;

export const Container = styled.div`
  min-height: 68px;
  width: 100%;
  border-right: 1px solid #cdd7df;
  border-left: 1px solid #cdd7df;
  padding: 0px 16px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

export const GreyFilterTag = styled(ModalTag)`
  font-weight: 400;
`;

export const Label = styled.div`
  font-family: Roboto;
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: left;
  color: #405f71;
  margin-right: 16px;
`;

export const Footer = styled(CardFooter)`
  width: 100%;
  min-height: 52px;
  height: 52px;
  background-color: #ffffff;
  border-radius: 0px 0px 4px 4px;
  padding-top: unset;
  padding-bottom: unset;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: row-reverse;
`;

export const ClearButton = styled(Button).attrs(() => ({ version: '2' }))`
  font-weight: 400;
`;

export const FilterButton = styled(Button).attrs(() => ({ version: '2' }))`
  font-weight: 400;
  color: ${getTheme('colors.liber', '#009dff')};
`;

export const Field = styled(TextField)``;

export const CleanAll = styled(FilterButton)`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
`;

export const Filtered = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: right;
  color: #405f71;
  margin-left: auto;
`;
