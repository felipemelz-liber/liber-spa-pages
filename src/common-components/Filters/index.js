export { default as ArrayFieldFilter } from './ArrayFieldFilter/ArrayFieldFilter.jsx';
export { default as TextFieldFilter } from './TextFieldFilter/TextFieldFilter.jsx';
export { default as DateFieldFilter } from './DateFieldFilter/DateFieldFilter.jsx';
