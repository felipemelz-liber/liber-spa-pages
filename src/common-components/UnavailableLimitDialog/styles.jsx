import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
