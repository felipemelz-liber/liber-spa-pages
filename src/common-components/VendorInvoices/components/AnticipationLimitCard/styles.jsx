import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
