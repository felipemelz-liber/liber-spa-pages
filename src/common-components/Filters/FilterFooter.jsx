import React from 'react';
import PropTypes from 'prop-types';
import { Footer, FilterButton, ClearButton } from './Filters.styles';

const FilterFooter = ({ disabled, onFilter, onClear }) => {
  return (
    <Footer>
      <FilterButton disabled={disabled} onClick={onFilter}>
        FILTRAR
      </FilterButton>
      <ClearButton onClick={onClear} disabled={disabled}>
        LIMPAR
      </ClearButton>
    </Footer>
  );
};

FilterFooter.propTypes = {
  disabled: PropTypes.bool,
  onFilter: PropTypes.func,
  onClear: PropTypes.func,
};

FilterFooter.defaultProps = {
  disabled: false,
  onFilter: () => null,
  onClear: () => null,
};

export default FilterFooter;
