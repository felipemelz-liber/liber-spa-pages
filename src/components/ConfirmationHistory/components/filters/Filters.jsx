import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Label, CleanAll, Filtered } from './Filters.styles.jsx';
import { DateFieldFilter } from '../../../../common-components/Filters/index.js';

const Filters = ({ count, handleFilter }) => {
  const [rangeField, setRangeField] = useState({
    start: null,
    end: null,
  });

  const handleClear = () => {
    const filters = {
      start: null,
      end: null,
    };
    setRangeField(filters);
    handleFilter(filters);
  };

  const handleDateFilter = rangeData => {
    setRangeField(rangeData);
    handleFilter(rangeData);
  };

  const { start, end } = rangeField;

  const renderOnFilter = () => {
    if (start || end) {
      return (
        <>
          <CleanAll onClick={handleClear}>Limpar filtros</CleanAll>
          <Filtered>{`${count} ${
            count === 1 ? 'confirmação filtrada' : 'confirmações filtradas'
          }`}</Filtered>
        </>
      );
    }
    return null;
  };

  return (
    <>
      <Label>Filtrar por</Label>
      <DateFieldFilter
        label="Data de Criação"
        description="Selecione uma data inicial e uma data final para o período de criação"
        onClear={handleClear}
        onFilter={handleDateFilter}
        selectedItems={[start, end]}
      />
      {renderOnFilter()}
    </>
  );
};

Filters.propTypes = {
  count: PropTypes.number.isRequired,
  handleFilter: PropTypes.func.isRequired,
};

export default Filters;
