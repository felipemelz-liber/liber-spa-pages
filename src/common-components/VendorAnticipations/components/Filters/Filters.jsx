import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DateFieldFilter from '../../../Filters/DateFieldFilter/DateFieldFilter';
import { REQUESTED_LT, REQUESTED_GT, STATUS, STATUS_IN } from '../../utils';
import { updateFilter } from '../../actions/VendorAdvancesActions';
import OptionsFilter from '../../../Filters/OptionsFilter/OptionsFilter';
import { FilterText, CleanFilters } from './Filters.styles';
import { dispatchDataLayerEvent } from '../../../../vendor/Utils';

const clearDateFilters = {
  [REQUESTED_LT]: null,
  [REQUESTED_GT]: null,
};

const clearStatusFilters = {
  [STATUS_IN]: null,
};

const allClearFilters = {
  ...clearDateFilters,
  ...clearStatusFilters,
};

const onDateFilter = (changeFilters, applyedFilters, isAdmin) => range => {
  dispatchDataLayerEvent('antecipacoesFiltrarSolicitacao', isAdmin);

  let filters = {};

  if (range && range.end) {
    filters = {
      [REQUESTED_GT]: range.start.format(),
      [REQUESTED_LT]: range.end.format(),
    };
  } else {
    filters = {
      [REQUESTED_GT]: range && range.start ? range.start.startOf('day').format() : null,
      [REQUESTED_LT]: range && range.start ? range.start.endOf('day').format() : null,
    };
  }

  changeFilters({ ...applyedFilters, ...filters });
};

const onStatusFilter = (filters, changeFilters, selectedStatuses, isAdmin) => {
  const status = selectedStatuses
    .map(statusId => (statusId === STATUS.finished ? 'Concluida' : 'Cancelada'))
    .join(' | ');

  dispatchDataLayerEvent('antecipacoesFiltrarStatus', isAdmin, {
    filtroStatusAntecipacoes: status,
  });

  changeFilters({ ...filters, [STATUS_IN]: selectedStatuses });
};

const handleDateFilterClear = (changeFilters, filters, isAdmin) => {
  dispatchDataLayerEvent('antecipacoesLimparFiltro', isAdmin);

  changeFilters({ ...filters, ...clearDateFilters });
};

const handleStatusFilterClear = (changeFilters, filters, isAdmin) => {
  dispatchDataLayerEvent('antecipacoesLimparFiltro', isAdmin);

  changeFilters({ ...filters, ...clearStatusFilters });
};

const handleClearAllFiltersClick = (changeFilters, isAdmin) => {
  dispatchDataLayerEvent('antecipacoesLimparFiltro', isAdmin);

  changeFilters(allClearFilters);
};

const hasFilters = filters =>
  Object.keys(filters).reduce((flag, key) => flag || Boolean(filters[key]), false);

export const Filters = ({ changeFilters, filters, isAdmin }) => {
  return (
    <>
      <FilterText>Filtrar por</FilterText>
      <DateFieldFilter
        label="Solicitação"
        onFilter={onDateFilter(changeFilters, filters, isAdmin)}
        onClear={() => handleDateFilterClear(changeFilters, filters, clearDateFilters, isAdmin)}
        selectedItems={[filters[REQUESTED_GT], filters[REQUESTED_LT]]}
      />
      <OptionsFilter
        filterValues={filters[STATUS_IN]}
        onFilter={selectedStatuses =>
          onStatusFilter(filters, changeFilters, selectedStatuses, isAdmin)
        }
        onClear={() => handleStatusFilterClear(changeFilters, filters, clearStatusFilters, isAdmin)}
        options={[
          { option: 'Concluída', value: STATUS.finished },
          { option: 'Cancelada', value: STATUS.canceled },
        ]}
        label="Status"
      />
      {hasFilters(filters) ? (
        <CleanFilters onClick={() => handleClearAllFiltersClick(changeFilters, isAdmin)}>
          Limpar filtros
        </CleanFilters>
      ) : null}
    </>
  );
};

Filters.propTypes = {
  changeFilters: PropTypes.func,
  filters: PropTypes.shape({
    [REQUESTED_LT]: PropTypes.string,
    [REQUESTED_GT]: PropTypes.string,
    [STATUS_IN]: PropTypes.arrayOf(PropTypes.number),
  }),
  isAdmin: PropTypes.bool,
};

Filters.defaultProps = {
  filters: {},
  isAdmin: false,
  changeFilters: () => {},
};

const mapStateToProps = ({ vendorAdvances: { filters, isAdmin } }) => ({ filters, isAdmin });

const mapDispatchToProps = {
  changeFilters: updateFilter,
};

export default connect(mapStateToProps, mapDispatchToProps)(Filters);
