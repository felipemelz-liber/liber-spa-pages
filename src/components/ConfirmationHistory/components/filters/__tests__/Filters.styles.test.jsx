import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import * as styles from '../Filters.styles';

describe('Testing Filters Styles', () => {
  configure({ adapter: new Adapter() });
  const { FilterTag, SelectedSwitch, ...uncontrolled } = styles;
  describe('Uncontrolled', () => {
    Object.keys(uncontrolled).forEach(style => {
      it(`renders ${style} correctly`, () => {
        const App = uncontrolled[style];
        const wrapper = shallow(<App />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
  describe('Controlled', () => {
    describe('FilterTag', () => {
      it('renders correctly without props', () => {
        const wrapper = shallow(<FilterTag />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly with gray=false', () => {
        const wrapper = shallow(<FilterTag gray={false} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly with gray=true', () => {
        const wrapper = shallow(<FilterTag gray />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
    describe('SelectedSwitch', () => {
      it('renders correctly without props', () => {
        const wrapper = shallow(<SelectedSwitch />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly with switch=false', () => {
        const wrapper = shallow(<SelectedSwitch switch={false} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly with switch=true', () => {
        const wrapper = shallow(<FilterTag switch />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
