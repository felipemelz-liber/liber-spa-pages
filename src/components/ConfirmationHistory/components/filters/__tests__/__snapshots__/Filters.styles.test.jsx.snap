// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Filters Styles Controlled FilterTag renders correctly with gray=false 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-weight: 400;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(ModalTag)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gray={false}
/>
`;

exports[`Testing Filters Styles Controlled FilterTag renders correctly with gray=true 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-weight: 400;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(ModalTag)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gray={true}
/>
`;

exports[`Testing Filters Styles Controlled FilterTag renders correctly without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-weight: 400;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(ModalTag)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Controlled SelectedSwitch renders correctly with switch=false 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  height: 52px;
  width: 100%;
  display: flex;
  flex-direction: column;
  transform: ",
          [Function],
          ";
  transition: transform 250ms ease-in-out;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  switch={false}
/>
`;

exports[`Testing Filters Styles Controlled SelectedSwitch renders correctly with switch=true 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-weight: 400;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(ModalTag)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  switch={true}
/>
`;

exports[`Testing Filters Styles Controlled SelectedSwitch renders correctly without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  height: 52px;
  width: 100%;
  display: flex;
  flex-direction: column;
  transform: ",
          [Function],
          ";
  transition: transform 250ms ease-in-out;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders CleanAll correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-iwsKbI",
        "isStatic": false,
        "rules": Array [
          "
  font-weight: 400;
  color: ",
          "
    ",
          [Function],
          "
  ",
          ";
",
          "
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
",
        ],
      },
      "displayName": "Styled(Styled(Buttons__Button))",
      "foldedComponentIds": Array [
        "Buttons__Button-u35aut-0",
        "sc-htoDjs",
      ],
      "render": [Function],
      "styledComponentId": "sc-iwsKbI",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders ClearButton correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": false,
        "rules": Array [
          "
  font-weight: 400;
",
        ],
      },
      "displayName": "Styled(Buttons__Button)",
      "foldedComponentIds": Array [
        "Buttons__Button-u35aut-0",
      ],
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders Field correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-dnqmqq",
        "isStatic": false,
        "rules": Array [],
      },
      "displayName": "Styled(Component)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-dnqmqq",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders FilterButton correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htoDjs",
        "isStatic": false,
        "rules": Array [
          "
  font-weight: 400;
  color: ",
          "
    ",
          [Function],
          "
  ",
          ";
",
        ],
      },
      "displayName": "Styled(Buttons__Button)",
      "foldedComponentIds": Array [
        "Buttons__Button-u35aut-0",
      ],
      "render": [Function],
      "styledComponentId": "sc-htoDjs",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders Filtered correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gZMcBi",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: right;
  color: #405f71;
  margin-left: auto;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gZMcBi",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders Footer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
  min-height: 52px;
  height: 52px;
  background-color: #ffffff;
  border-radius: 0px 0px 4px 4px;
  padding-top: unset;
  padding-bottom: unset;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: row-reverse;
",
        ],
      },
      "displayName": "Styled(Cards__CardFooter)",
      "foldedComponentIds": Array [
        "Cards__CardFooter-sc-1kn0psw-3",
      ],
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders GreyFilterTag correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": false,
        "rules": Array [
          "
  font-weight: 400;
",
        ],
      },
      "displayName": "Styled(ModalTag)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders Label correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.1px;
  text-align: left;
  color: #405f71;
  margin-right: 16px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders SelectedLabelContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Filters Styles Uncontrolled renders SelectedLabelOption correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
  height: 26px;
  display: flex;
  justify-content: center;
  align-items: center;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
